sros/docker/launch.py

```diff
@@ -35,23 +35,47 @@ def trace(self, message, *args, **kws):
 
 logging.Logger.trace = trace
 
+# Default power type to use, SROS complains
+# "Class Power Shelf : wrong type inserted" when using "ac/hv" here
+DEFAULT_POWER = "dc"
+
+def LINE_CARD(chassis,card,mda,integrated=False,card_type=None):
+    """
+    Configures a line card in a distributed or integrated deployment model.
+    Note that it may be possible to omit configuring these explicitly
+    """
+    slot = "A" if integrated else "1"
+    return {
+      "timos_line": f"slot={slot} chassis={chassis} card={card} mda/1={mda}",
+      "card_config": f"""
+       /configure card 1 card-type {card_type if card_type else card}
+       /configure card 1 mda 1 mda-type {mda}
+      """,
+    }
+
 SROS_VARIANTS = {
     "ixr-e-big": {
         "deployment_model": "distributed",
         # control plane (CPM)
-        "max_nics": 34,
+        "max_nics": 34, # 24*10 + 8*25G + 2*100G (with connector)
         "cp": {
             "min_ram": 3,
-            "timos_line": "slot=A chassis=ixr-e card=cpm-ixr-e",
+            "timos_line": "slot=A chassis=ixr-e card=cpm-ixr-e/imm24-sfp++8-sfp28+2-qsfp28",
```

Comment:
should the cpm of a control plane vm specify the imm?
copying @LimeHat

Reply:
Not needed AFAIK, I've been using this in the past:
```
               slot=A \
                           chassis=IXR-e \
                           card=cpm-ixr-e
```